### PR TITLE
User CRUD

### PR DESCRIPTION
**Pull request by satriowisnugroho**
This PR creates:

- GET `/users`
- POST `/users`
- GET `/users/:id`
- PUT `/users/:id`
- DELETE `/users/:id`

User collection https://www.getpostman.com/collections/518a575088c8431b7bf3

### CODE REVIEW

File: controllers/user.js [hunk 1]
```diff
@@ -409,3 +410,88 @@ exports.getUsersDatatable = (req, res) => {
...
+    name: req.body.name,
+    email: req.body.email,
+    role: req.body.role,
+    password: rand,
+  }).save((err) => {
+    if (err) {
+      return next(err);
+    }
+    req.flash('success', { msg: 'New user has been created.' });
+    res.redirect('/dashboard/users');
```
Comment (haydarai) [inline controllers/user.js]: This always give me response `Cannot GET /dashboard/users`

File: controllers/user.js [hunk 2]
```diff
@@ -409,3 +410,88 @@ exports.getUsersDatatable = (req, res) => {
...
+/**
+ * GET /users/:id
+ * Get user data
+ */
+exports.getUser = (req, res, next) => {
+  User.findById(req.params.id, (err, user) => {
+    if (err) {
+      return next(err);
+    }
+    res.json(user);
```
Comment (haydarai) [inline controllers/user.js]: Please encapsulate it into `{user}`

File: controllers/user.js [hunk 3]
```diff
@@ -409,3 +410,88 @@ exports.getUsersDatatable = (req, res) => {
...
+/**
+ * PUT /users/:id
+ * Update user data
+ */
+exports.putUser = (req, res, next) => {
+  User.findByIdAndUpdate(req.params.id, req.body, { new: true }, (err, user) => {
+    if (err) {
+      return next(err);
+    }
+    res.json(user);
```
Comment (haydarai) [inline controllers/user.js]: Please encapsulate it into {user}

**Comment by satriowisnugroho** [comment]
ok thank you @haydarai . i'll fix it later

File: controllers/user.js (hunk 1 above)
Comment (dwiajik) [reply controllers/user.js]: please redirect to /users

File: controllers/user.js [hunk 4]
```diff
@@ -2,6 +2,7 @@ const bluebird = require('bluebird');
 const crypto = bluebird.promisifyAll(require('crypto'));
 const nodemailer = require('nodemailer');
 const passport = require('passport');
+const randomString = require('randomstring');
```
Comment (dwiajik) [inline controllers/user.js]: I think it's better to use faker instead, because it's already used https://github.com/Marak/faker.js/wiki/Internet

**Comment by haydarai** [comment]
@dwiajik I've fixed this PR, can you please re-check it once more?

**Comment by satriowisnugroho** [comment]
wow it's closed -_-

**Comment by haydarai** [comment]
lol, because it's just minor changes so I thought I'll fix it myself 😂

**Pull request merged**